### PR TITLE
Add preliminary BookList class.

### LINKED ISSUE #101: Create a BookList class
Create a class to encapsulate a list of books.

### PR DESCRIPTION
**Pull request by Edfernape**
A simple BookList class.

Resolves #101

### CODE REVIEW

File: src/main/java/bookface/logic/commands/add/AddBookCommand.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package bookface.logic.commands.add;
+
+import static bookface.logic.parser.CliSyntax.PREFIX_AUTHOR;
+import static bookface.logic.parser.CliSyntax.PREFIX_TITLE;
```
Comment (TYKCodes) [inline src/main/java/bookface/logic/commands/add/AddBookCommand.java:4]: Are these prefixes created in the Clisyntax class?

File: src/main/java/bookface/logic/commands/add/AddBookCommand.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * The command that adds a book to the BookList.
+ */
+public class AddBookCommand extends AddCommand {
+    public static final String COMMAND_WORD = "book";
+    public static final String MESSAGE_USAGE = AddCommand.COMMAND_WORD
+            + " " + COMMAND_WORD
+            + ": Adds a book to the book list."
+            + "Parameters: "
+            + PREFIX_AUTHOR + "AUTHOR "
+            + PREFIX_TITLE + "TITLE \n"
```
Comment (TYKCodes) [inline src/main/java/bookface/logic/commands/add/AddBookCommand.java:22]: Perhaps the order of the parameters should be changed here to match the order of the command fields as well?

File: src/main/java/bookface/logic/commands/add/AddBookCommand.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            + "Example: " + AddCommand.COMMAND_WORD
+            + " " + COMMAND_WORD + " "
+            + PREFIX_TITLE + "The Hobbit "
+            + PREFIX_AUTHOR + "J.R.R. Tolkien";
+    public static final String MESSAGE_SUCCESS = "New book added: %1$s";
+    public static final String MESSAGE_DUPLICATE_BOOK = "This book is already in our records.";
+    private final Book book;
```
Comment (TYKCodes) [inline src/main/java/bookface/logic/commands/add/AddBookCommand.java:29]: Maybe we can adapt the variable name from AddUserCommand as well? Something like this:
`private final Book toAdd`

Subsequent variable name instances should follow suit if changed

File: src/main/java/bookface/logic/commands/add/AddBookCommand.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        requireNonNull(book);
+        this.book = book;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        model.addBook(this.book);
+        return new CommandResult(String.format(MESSAGE_SUCCESS, this.book));
+    }
```
Comment (TYKCodes) [inline src/main/java/bookface/logic/commands/add/AddBookCommand.java:45]: Is this method missing a check to ensure the Book does not already exist in the records?
Perhaps the `hasPerson` method needs to be implemented to perform this check

**Review by TYKCodes — CHANGES_REQUESTED** [review]
I think there are a few missing codes that need to be added/created prior to the usage of this class. We can go through them whenever you are available

File: src/main/java/bookface/logic/commands/add/AddBookCommand.java (hunk 4 above)
Comment (Edfernape) [reply src/main/java/bookface/logic/commands/add/AddBookCommand.java:45]: ohh I was thinking that we should allow duplicate books and then add functionality for duplicates later on. Can, I'll make the booklist accept unique books for now

File: src/main/java/bookface/logic/commands/add/AddBookCommand.java (hunk 3 above)
Comment (Edfernape) [reply src/main/java/bookface/logic/commands/add/AddBookCommand.java:29]: i understand, I'll make the change

File: src/test/java/bookface/logic/commands/add/AddUserCommandTest.java [hunk 5]
```diff
@@ -10,6 +10,7 @@
 import java.util.Arrays;
 import java.util.function.Predicate;
 
+import bookface.model.book.Book;
 import org.junit.jupiter.api.Test;
 
 import bookface.commons.core.GuiSettings;
```
[the remark this answers is not in the document]
Comment (Edfernape) [reply src/test/java/bookface/logic/commands/add/AddUserCommandTest.java]: fixed, thanks

File: src/main/java/bookface/logic/commands/add/AddBookCommand.java (hunk 1 above)
Comment (Edfernape) [reply src/main/java/bookface/logic/commands/add/AddBookCommand.java:4]: yup!

File: src/main/java/bookface/logic/commands/add/AddBookCommand.java (hunk 2 above)
Comment (Edfernape) [reply src/main/java/bookface/logic/commands/add/AddBookCommand.java:22]: my bad, fixed now

File: src/main/java/bookface/logic/commands/add/AddBookCommand.java (hunk 3 above)
Comment (Edfernape) [reply src/main/java/bookface/logic/commands/add/AddBookCommand.java:29]: changed it to bookToAdd so it's clearer

**Comment by Edfernape** [comment]
@TYKCodes thanks for the review! just committed the changes

**Pull request merged**